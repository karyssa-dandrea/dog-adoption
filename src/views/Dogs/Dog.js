import React from 'react';
import { fetchDogById } from '../../services/dogs';
import { useState, useEffect } from 'react';
import DogDetail from '../../components/DogDeets/DogDetails';

export default function Dog(props) {
  const [dog, setDog] = useState([]);
  const [loading, setLoading] = useState(true);
  const id = props.match.params.id;

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchDogById(id);
      setDog(data);
      setLoading(false);
    };
    fetchData();
  }, [id]);

  if (loading) {
    return <h3>Loading</h3>;
  }

  return (
    <div className="dog-deets">
      <DogDetail dog={dog} />
    </div>
  );
}
