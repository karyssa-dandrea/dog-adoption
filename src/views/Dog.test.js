import { render, screen } from '@testing-library/react';
import Dog from '../views/Dogs/Dog';
import { MemoryRouter } from 'react-router-dom';

test('should render dog details', async () => {
  const { container } = render(
    <MemoryRouter>
      <Dog match={{ params: { id: 2 } }} />
    </MemoryRouter>
  );
  await screen.findByText('Amely');
  expect(container).toMatchSnapshot();
});
