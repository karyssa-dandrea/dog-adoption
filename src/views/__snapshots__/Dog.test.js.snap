// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render dog details 1`] = `
<div>
  <div
    class="dog-deets"
  >
    <div
      class="dog-details"
    >
      <h1>
        Amely
      </h1>
      <img
        src="https://placedog.net/500?id=12"
      />
      <p>
         
        Amely
         is a(n) 
        0
         year old 
        Rottweiler
      </p>
      <p>
         
        Highly energetic - loves long walks and tug of war. Expect he'll grow to be a lot bigger than a teacup.
      </p>
    </div>
  </div>
</div>
`;
